src/test/jtx/mpt.h

```diff
@@ -0,0 +1,267 @@
+    std::optional<MPTID> id = std::nullopt;
+    std::optional<std::uint32_t> ownerCount = std::nullopt;
+    std::optional<std::uint32_t> holderCount = std::nullopt;
+    std::optional<std::uint32_t> flags = std::nullopt;
+    std::optional<TER> err = std::nullopt;
+};
+
+struct MPTSet
+{
+    AccountP account = nullptr;
+    AccountP holder = nullptr;
+    std::optional<MPTID> id = std::nullopt;
+    std::optional<std::uint32_t> ownerCount = std::nullopt;
+    std::optional<std::uint32_t> holderCount = std::nullopt;
+    std::optional<std::uint32_t> flags = std::nullopt;
+    std::optional<TER> err = std::nullopt;
+};
+
+class MPTTester
+{
+    Env& env_;
+    Account const& issuer_;
+    std::unordered_map<std::string, AccountP> const holders_;
+    std::optional<MPTID> id_;
+    std::optional<uint256> issuanceKey_;
+    bool close_;
+
+public:
+    MPTTester(Env& env, Account const& issuer, MPTConstr const& constr = {});
+
+    void
+    create(MPTCreate const& arg = MPTCreate{});
+
+    void
+    destroy(MPTDestroy const& arg = MPTDestroy{});
+
+    void
+    authorize(MPTAuthorize const& arg = MPTAuthorize{});
+
+    void
+    set(MPTSet const& set = {});
+
+    [[nodiscard]] bool
+    checkMPTokenAmount(Account const& holder, std::int64_t expectedAmount)
+        const;
+
+    [[nodiscard]] bool
+    checkMPTokenOutstandingAmount(std::int64_t expectedAmount) const;
+
+    [[nodiscard]] bool
+    checkFlags(uint32_t const expectedFlags, AccountP holder = nullptr) const;
+
+    Account const&
+    issuer() const
+    {
+        return issuer_;
+    }
+    Account const&
+    holder(std::string const& h) const;
+
+    void
+    pay(Account const& src,
+        Account const& dest,
+        std::int64_t amount,
+        std::optional<TER> err = std::nullopt);
+
+    void
+    claw(
+        Account const& issuer,
+        Account const& holder,
+        std::int64_t amount,
+        std::optional<TER> err = std::nullopt);
+
+    PrettyAmount
+    mpt(std::int64_t amount) const;
+
+    uint256 const&
+    issuanceKey() const
+    {
+        assert(issuanceKey_);
+        return *issuanceKey_;
+    }
+
+    MPTID const&
+    issuanceID() const
+    {
+        assert(id_);
+        return *id_;
+    }
+
+    std::int64_t
+    getAmount(Account const& account) const;
+
+    MPT
+    operator[](std::string const& name);
+
+private:
+    using SLEP = std::shared_ptr<SLE const>;
+    bool
+    forObject(
+        std::function<bool(SLEP const& sle)> const& cb,
+        AccountP holder = nullptr) const;
+
+    template <typename A>
+    TER
+    submit(A const& arg, Json::Value const& jv)
+    {
+        if (arg.err)
+        {
+            if (arg.flags && arg.flags > 0)
+                env_(jv, txflags(*arg.flags), ter(*arg.err));
+            else
+                env_(jv, ter(*arg.err));
+        }
+        else if (arg.flags && arg.flags > 0)
+            env_(jv, txflags(*arg.flags));
+        else
+            env_(jv);
```

Comment:
```suggestion
        auto const expected = arg.err.value_or(tesSUCCESS);
        if (arg.flags && arg.flags > 0)
            env_(jv, txflags(*arg.flags), ter(expected));
        else
            env_(jv, ter(*arg.err));
```
You could probably do something similar with `arg.flags`.